packages/eslint-plugin/src/rules/no-misused-spread.ts

```diff
@@ -0,0 +1,47 @@
+import {
+  ESLintUtils,
+  type TSESLint,
+  type TSESTree,
+} from '@typescript-eslint/utils';
+
+import { createRule } from '../util';
+
+export default createRule({
+  defaultOptions: [],
+  name: 'no-misused-spread',
+  meta: {
+    docs: {
+      description: 'Disallow spread operator on function',
+      recommended: 'stylistic',
+      requiresTypeChecking: true,
+    },
+    messages: {
+      forbidden:
```

Comment:
in case we'll expand this rule in the future, maybe change the message id?
(just something from top of my head, not sure it's a good one though)

```suggestion
      forbiddenFunctionSpread:
```

Reply:
It makes sense.